File: ml-services/standardize/models/transaction.py
from pydantic import BaseModel
from typing import Optional


class StandardizedTransaction(
    BaseModel
):

    date: Optional[str] = None

    amount: Optional[float] = None

    txn_type: Optional[str] = None

    reference_number: Optional[str] = None

    narration: Optional[str] = None

    narration_normalized: Optional[str] = None

    balance: Optional[float] = None

    debit_credit: Optional[str] = None

    platform: Optional[str] = None

    upi_id: Optional[str] = None

    # -------------------------
    # Investigation Dataset
    # -------------------------

    sender_account: Optional[str] = None

    receiver_account: Optional[str] = None

    bank_name: Optional[str] = None

File: backend/src/models/transaction.rs
use serde::{
    Deserialize,
    Serialize,
};

#[derive(
    Debug,
    Serialize,
    Deserialize,
)]
pub struct Transaction {

    pub date: Option<String>,

    pub amount: Option<f64>,

    pub txn_type: Option<String>,

    pub reference_number:
        Option<String>,

    pub narration:
        Option<String>,

    pub narration_normalized:
        Option<String>,

    pub balance:
        Option<f64>,

    pub debit_credit:
        Option<String>,

    pub platform:
        Option<String>,

    pub upi_id:
        Option<String>,

    pub sender_account:
    Option<String>,

    pub receiver_account:
        Option<String>,

    pub bank_name:
        Option<String>,

    pub is_duplicate:
        Option<bool>,

    pub is_failed:
        Option<bool>,

    pub is_valid:
        Option<bool>,

    pub confidence_score:
        Option<f64>,

    pub validation_notes:
        Option<Vec<String>>,
}